sale_force_invoiced/model/sale_order.py

```diff
@@ -8,20 +7,19 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
-    force_invoiced = fields.Boolean(string='Force invoiced',
-                                    help='When you set this field, the sales '
-                                         'order will be considered as fully '
-                                         'invoiced, even when there may be '
-                                         'ordered or delivered quantities '
-                                         'pending to invoice.',
-                                    readonly=True,
-                                    states={'done': [('readonly', False)]},
-                                    copy=False,
-                                    )
+    force_invoiced = fields.Boolean(
+        string='Force invoiced',
+        help='When you set this field, the sales order will be considered as '
+             'fully invoiced, even when there may be ordered or delivered '
+             'quantities pending to invoice.',
+        readonly=True,
+        states={'done': [('readonly', False)]},
+        copy=False,
+    )
 
     @api.depends('force_invoiced')
     def _get_invoiced(self):
         super(SaleOrder, self)._get_invoiced()
-        for order in self:
-            if order.force_invoiced and order.invoice_status == 'to invoice':
-                order.invoice_status = 'invoiced'
+        self.filtered(lambda so: so.force_invoiced
+                      and so.invoice_status == 'to invoice').write(
+            {'invoice_status': 'invoiced'})
```

Comment:
Revert this change, you cannot use write inside an api.depends.

Reply:
done